### PR TITLE
APIS-7300 - Add option for generating interface GraphQL implementations in separate files.

### PR DESCRIPTION
**Pull request by gannett-ggreer**

### CODE REVIEW

File: generate/graphql.go [hunk 1]
```diff
@@ -200,9 +200,31 @@ func buildGraphQLFile(schemaPath, name, packageName string, args BuildArgs) erro
 			generated.rootStruct.fields[lf] = lfd[lf]
 			generated.rootStruct.fieldOrder = append(generated.rootStruct.fieldOrder, lf)
 		}
+
+		gfile := gfile
```
Comment (liamphmurphy) [inline generate/graphql.go:204]: As a potential nit, should this be `gFile`?

File: generate/graphql.go [hunk 2]
```diff
@@ -200,9 +200,31 @@ func buildGraphQLFile(schemaPath, name, packageName string, args BuildArgs) erro
...
+			if !strings.HasPrefix(oneOfName, name) {
+				outName = name + "_" + outName
+			}
+			outPath = filepath.Join(args.OutputDirGraphQL, outName)
+			gfile, err = os.Create(outPath)
+			if err != nil {
+				return fmt.Errorf("failed to open file %q: %v", outPath, err)
+			}
+			if _, err := fmt.Fprintf(gfile, "# %s\n", disclaimer); err != nil {
+				return fmt.Errorf("failed writing GraphQL: %v", err)
```
Comment (liamphmurphy) [inline generate/graphql.go:217]: `%w` for these errors instead of `%v`?

File: generate/graphql.go [hunk 3]
```diff
@@ -200,9 +200,31 @@ func buildGraphQLFile(schemaPath, name, packageName string, args BuildArgs) erro
...
+			}
+		}
 		if err := generated.write(gfile); err != nil {
 			return fmt.Errorf("failed to write oneOf struct: %v", err)
 		}
+		if args.InterfaceFiles {
+			if err := gfile.Close(); err != nil {
+				return fmt.Errorf("failed to close file %q: %v", outPath, err)
```
Comment (liamphmurphy) [inline generate/graphql.go:225]: Same `%w` here.

File: generate/graphql.go (hunk 1 above)
Comment (gannett-ggreer) [reply generate/graphql.go:204]: Strictly speaking, yes, however that's the name of the variable from the existing (`generate.go`) code so I kept it like that.

File: generate/graphql.go (hunk 2 above)
Comment (gannett-ggreer) [reply generate/graphql.go:217]: These I can at least change to match current proper practices.

**Pull request merged**